cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,211 @@
+                    if (verified_partition_group_access
+                            and verified_course_user_group_partition_group.group_id in verified_partition_group_access):
+                        set_verified_enrollment_track = True
+
+                    # Add the enrollment track ids to a group access array
+                    enrollment_track_group_access = []
+                    if set_audit_enrollment_track:
+                        enrollment_track_group_access.append(settings.COURSE_ENROLLMENT_MODES['audit'])
+                    if set_verified_enrollment_track:
+                        enrollment_track_group_access.append(settings.COURSE_ENROLLMENT_MODES['verified'])
+
+                    # If either the audit track, or verified track needed an update, set the access, update and publish
+                    if set_verified_enrollment_track or set_audit_enrollment_track:
+                        # Sets whether or not an xblock has changes
+                        has_changes = module_store.has_changes(item)
+
+                        # Check that the xblock does not have changes and add it to be updated, otherwise add an error
+                        if not has_changes:
+                            item.group_access = {ENROLLMENT_TRACK_PARTITION_ID: enrollment_track_group_access}
+                            items_to_update.append(item)
+                        else:
+                            errors.append("XBlock '%s' with location '%s' needs access changes, but is a draft"
+                                          % (item.display_name, item.location))
+
+            partitions_to_delete = random_audit_course_user_group_partition_groups
+            partitions_to_delete.append(verified_course_user_group_partition_group)
+
+            # If there are no errors iterate over and update all of the items that had the access changed
+            if not errors:
+                for item in items_to_update:
+                    module_store.update_item(item, ModuleStoreEnum.UserID.mgmt_command)
+                    module_store.publish(item.location, ModuleStoreEnum.UserID.mgmt_command)
+
+            # Check if we should delete any partition groups if there are no errors.
+            # If there are errors, none of the xblock items will have been updated,
+            # so this section will throw errors for each partition in use
+            if partitions_to_delete and not errors:
+                partition_service = PartitionService(rerun_course_key)
+                course = partition_service.get_course()
+                for partition_to_delete in partitions_to_delete:
+                    # Get the user partition, and the index of that partition in the course
+                    partition = partition_service.get_user_partition(partition_to_delete.partition_id)
+                    partition_index = course.user_partitions.index(partition)
+                    group_id = int(partition_to_delete.group_id)
```

Comment:
I'm confused about what these structures look like. How do we know there is a single group_id in partition_to_delete?